src/main/java/org/gridsuite/study/server/StudyController.java

```diff
@@ -540,4 +546,17 @@ public StudyController(StudyService studyService) {
     public ResponseEntity<Mono<Void>> stopSecurityAnalysis(@ApiParam(value = "Study name") @PathVariable("studyUuid") UUID studyUuid) {
         return ResponseEntity.ok().body(studyService.stopSecurityAnalysis(studyUuid));
     }
+
+    @GetMapping(value = "/studies/{studyUuid}/network-map/{equipmentType}/{equipmentId}")
+    @ApiOperation(value = "Get Network equipment description", produces = "application/json")
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "The equipment data")})
+    public ResponseEntity<Mono<String>> getEquipmentData(
+        @PathVariable("studyUuid") UUID studyUuid,
+        @PathVariable("equipmentType") EquipmentType equipmentType,
+        @PathVariable("equipmentId") String equipmentId) {
+
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.getNetworkUuid(studyUuid)
+            .flatMap(uuid -> studyService.getEquipmentData(uuid, equipmentType, equipmentId)));
```

Comment:
Here, equipmentType can be null, since the PathVariable "equipmentType" can be different from the accepted types of the enum (method fromString of the EquipmentType returns null if type is not one of the enum, and so EquipmentTypeEnumConverter setAsText method sets a null value). The case of a null value for equipmentType is not correctly addressed in the getEquipmentData method of the StudyService.

Reply:
nope : you get the following error with a bad type
org.springframework.web.server.ServerErrorException: 500 INTERNAL_SERVER_ERROR "equipmentType"

Reply:
Maybe throw a more explicit exception then ?